File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-0.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> stk;
        for (int i = 0; i < tokens.size(); i++){
            if (tokens[i] == "+" || tokens[i] == "-" || tokens[i] == "*" || tokens[i] == "/") {
                int right = stk.top();
                stk.pop();
                int left = stk.top();
                stk.pop();
                int res;
                if (tokens[i] == "+") res = left + right;
                else if (tokens[i] == "-") res = left - right;
                else if (tokens[i] == "*") res = left * right;
                else res = left / right;
                stk.push(res);
            } else {
                stk.push(stoi(tokens[i]));
            }
        }
        return stk.top();
    }
};

File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-0.py
class Solution:   
    def isSubtree(self, root: Optional[TreeNode], subRoot: Optional[TreeNode]) -> bool: 
        def isSameTree(p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
            if not p and not q:
                return True
            if not p or not q:
                return False

            if p.val != q.val:
                return False

            left = isSameTree(p.left, q.left)
            right = isSameTree(p.right, q.right)

            return left and right

        if not root:
            return False
        
        res = isSameTree(root, subRoot)
        
        return res or self.isSubtree(root.left, subRoot) or self.isSubtree(root.right, subRoot)